### PR TITLE
Authentication

### LINKED ISSUE #6: Adding User Authentication

### PR DESCRIPTION
**Pull request by UP419**

### CODE REVIEW

File: backend/src/api/auth_router.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+auth_router = APIRouter(prefix="/api/users")
+
+
+@auth_router.post("/add")
+@inject
+def add_user(user: user.UserCreate, auth_service: AuthService = Depends()):
+    return auth_service.add_user(user)
+
+
+@auth_router.get("/getUser")
```
Comment (AkakiZZ) [inline backend/src/api/auth_router.py]: Use kebab case -> /get-user or just /user /users

File: backend/src/service/auth_service.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        db_user = self.get_user_by_email(user.email)
+        if db_user:
+            raise HTTPException(status_code=400, detail="Email already registered")
+        db_user = self.get_user_by_username(user.username)
+        if db_user:
+            raise HTTPException(status_code=400, detail="Username taken")
+        return self.create_user(user)
+
+    def get_user(self, usernameOrEmail: str, password: str):
+        if '@' in usernameOrEmail:
```
Comment (AkakiZZ) [inline backend/src/service/auth_service.py:30]: Not reliable solution

File: docker-compose.yml [hunk 3]
```diff
@@ -31,4 +33,16 @@ services:
...
-      - ./webapp.env
\ No newline at end of file
+      - ./webapp.env
+
+  codemeet_db:
+    image: "postgres:14"
+    ports:
+      - "5432:5432"
+    environment:
+      - POSTGRES_USER=codemeet
```
Comment (AkakiZZ) [inline docker-compose.yml]: move those variables in .env file

File: frontend/src/pages/main-page/MainPage.tsx [hunk 4]
```diff
@@ -4,6 +4,7 @@ import openviduApi from "../../api/openvidu";
```
Comment (AkakiZZ) [inline frontend/src/pages/main-page/MainPage.tsx]: Rollback this file completely

File: backend/src/sql_app/main.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
+from fastapi import Depends, FastAPI, HTTPException
```
Comment (AkakiZZ) [inline backend/src/sql_app/main.py]: This file probably should be removed

File: backend/src/service/auth_service.py (hunk 2 above)
Comment (UP419) [reply backend/src/service/auth_service.py:30]: yes, its only temporary workaround

**Review by AkakiZZ — APPROVED** [review]

**Pull request merged**